### ISSUE TITLE
Global registrator support

### ISSUE THREAD

**Issue by fabiospampinato** [WebReflection/linkedom#193]
[Happy DOM](https://github.com/capricorn86/happy-dom/tree/master/packages/happy-dom) has a [global-registrator](https://github.com/capricorn86/happy-dom/tree/master/packages/global-registrator) package, that turns the server environment into a browser environment, to the extent that Happy DOM can do that. It'd be useful to have a similar package for `linkedom` too.

I tried doing this manually, but there are some problems with it:

1. `parseHTML` seems to return a proxy, so it's unclear which APIs are provided and which aren't. I'd have to manually find and list all of them.
2. Even if I do that, which seems pretty tedious, if new ones get added I'd have to update my code to use them, while ideally my could just work with any version of linkedom.
3. Some regular exports, like `atob`, should probably be added to `globalThis` only if they don't exist already. Handling details like that would probably be done best by linkedom itself. Plus there could also be new exports that require user code to be changed to use them, if there isn't a built-in global registrator.

**Comment by WebReflection** [comment]
I'm using LinkeDOM to test almost every library of mine ... and I am not sure what you are asking, or what it's missing ... but most importantly, if Happy DOM already serve you well, why should I copy that in here?

In any case, the JSDOM example in the readme already gives you a `window` and a `document` but it's true it doesn't expose much unless accessed because that's the nature of the module itself: lazy load anything.

In short: what is the use case here? And since Happy DOM looks like a LinkeDOM alternative, why not using Happy DOM when this use case matters? 🤔

**Comment by fabiospampinato** [comment]
> I'm using LinkeDOM to test almost every library of mine ... and I am not sure what you are asking, or what it's missing ...

I'm "asking" for a simple way to turn a Node environment into a browser environment with `linkedom`, like if I have this:

```ts
import './dom-shim';
```

What code would that module need to contain to work with `linkedom`? Assuming every provided API that makes sense to attach to `globalThis` is attached. 

Like, with Happy DOM i can just call the global registrator, with `linkedom` I need first of all to understand which APIs are provided by `parseHTML`, which doesn't seem obvious because it returns a Proxy.

> if Happy DOM already serve you well, why should I copy that in here?

Happy DOM is actually returning me something that looks incorrect, as in: executing the same code in the browser gives me a different output HTML string, so I wanted to try what the output was with `linkedom`, to double check. And JSDOM seems problematic to work with with my setup.

Besides I haven't really benchmarked the two, so I don't know which one would serve me better really.

> In short: what is the use case here? And since Happy DOM looks like a LinkeDOM alternative, why not using Happy DOM when this use case matters? 🤔

The use case is poor man's SSR by running client code in the server. Happy DOM seems to produce an incorrect output, though I'd like to double-check with another library before being "sure" of that.

In any case if this feature is out of scope no problem 👍

**Comment by fabiospampinato** [comment]
FWIW I just tried exporting things from `linkedom` until my code worked, and it seems to indeed produce a correct output, the same one I'm getting in the browser, which is not the one that Happy DOM is giving me.

**Comment by WebReflection** [comment]
for history sake: https://github.com/WebReflection/linkedom#whats-new
that satisfied a lot of use cases ... except the give me all windows keys and set these as global ... that one, I am afraid, is not available in here, but it looks like in Happy DOM it's a module a part, so maybe a module a part would be a better option for this?

**Comment by fabiospampinato** [comment]
Anyway I think I'm content enough to make a little `linkedom-global` package that does what I need manually. Feel free to re-open if you'd like to add something like that to `linkedom` itself.

Thanks for making `linkedom` ❤️

**Comment by fabiospampinato** [comment]
For context I ended up with this: https://github.com/fabiospampinato/linkedom-global
